radix-engine/src/updates/cuttlefish.rs

```diff
@@ -194,6 +217,167 @@ fn generate_cuttlefish_transaction_validation_updates() -> StateUpdates {
+            added_functions
+                .keys()
+                .map(ToOwned::to_owned)
+                .map(|ident| MethodKey { ident })
+                .map(|key| (key, MethodAccessibility::Public)),
+        );
+        auth_config.into_locked_substate()
+    };
+
+    // Updating the blueprint definition of the existing blueprint with the added functions.
+    let blueprint_definition_substate = {
+        let mut blueprint_definition = reader
+            .read_object_collection_entry::<_, VersionedPackageBlueprintVersionDefinition>(
+                &node_id,
+                ObjectModuleId::Main,
+                ObjectCollectionKey::KeyValue(
+                    PackageCollection::BlueprintVersionDefinitionKeyValue.collection_index(),
+                    &blueprint_version_key,
+                ),
+            )
+            .unwrap()
+            .unwrap()
+            .fully_update_and_into_latest_version();
+
+        for (function_name, added_function) in added_functions.into_iter() {
+            let TypeRef::Static(input_local_id) = added_function.input else {
+                unreachable!()
+            };
+            let TypeRef::Static(output_local_id) = added_function.output else {
+                unreachable!()
+            };
+
+            blueprint_definition.function_exports.insert(
+                function_name.clone(),
+                PackageExport {
+                    code_hash,
+                    export_name: function_name.clone(),
+                },
+            );
+            blueprint_definition.interface.functions.insert(
+                function_name,
+                FunctionSchema {
+                    receiver: added_function.receiver,
+                    input: BlueprintPayloadDef::Static(ScopedTypeId(schema_hash, input_local_id)),
+                    output: BlueprintPayloadDef::Static(ScopedTypeId(schema_hash, output_local_id)),
+                },
+            );
+        }
+
+        blueprint_definition.into_locked_substate()
+    };
+
+    // Getting the partition number of the various collections that we're updating
+    let [blueprint_version_definition_partition_number, code_vm_type_partition_number, code_original_code_partition_number, schema_partition_number, blueprint_version_auth_config_partition_number] =
+        [
+            PackageCollection::BlueprintVersionDefinitionKeyValue,
+            PackageCollection::CodeVmTypeKeyValue,
+            PackageCollection::CodeOriginalCodeKeyValue,
+            PackageCollection::SchemaKeyValue,
+            PackageCollection::BlueprintVersionAuthConfigKeyValue,
+        ]
+        .map(|package_collection| {
+            reader
+                .get_partition_of_collection(
+                    &node_id,
+                    ObjectModuleId::Main,
+                    package_collection.collection_index(),
+                )
+                .unwrap()
+        });
+
+    // Generating the state updates
+    StateUpdates {
```

Comment:
I find this hard to read. Can we use `StateUpdates::empty().set_substate(...)` instead?

Or better yet, to avoid setting the node id multiple times: `let node_state_updates = NodeStateUpdates::empty().set_substate(partition, key, value).set_substate(partition, key, value)` and then `StateUpdates::set_node(node_state_updates)`.

Reply:
I like that! I just made that change the code looks as follows now:

```rust
StateUpdates::empty()
    .set_node_updates(node_id, NodeStateUpdates::empty()
    .set_substate(
        blueprint_version_definition_partition_number,
        SubstateKey::map(blueprint_version_key),
        blueprint_definition_substate,
    )
    .set_substate(
        code_vm_type_partition_number,
        SubstateKey::map(code_hash),
        vm_type_substate,
    )
    .set_substate(
        code_original_code_partition_number,
        SubstateKey::map(code_hash),
        code_substate,
    )
    .set_substate(
        schema_partition_number,
        SubstateKey::map(schema_hash),
        schema_substate,
    )
    .set_substate(
        blueprint_version_auth_config_partition_number,
        SubstateKey::map(blueprint_version_key),
        auth_config,
    ))
```

Would love to get your opinion on it :)